### DISCUSSION TITLE
The NodeJS program I deploy using pm2 always stop running

### DISCUSSION THREAD

**Discussion by skirodev** [domcloud/domcloud#44] [Q&A] [answered]
Why does my NodeJS program deployed with pm2 always stop running every few hours? It's a tiny script and consumes less than 100MB of memory. 

I would be very grateful if you could help me

**Comment by willnode** [comment, marked as answer]
Long-running processes are [discouraged](https://domcloud.co/docs/features/service#ephemeral-ports:~:text=Please%20note%20that%20long%2Drunning%20background%20processes%20are%20not%20allowed%20in%20the%20server.%20It%20will%20be%20killed%20automatically.) and we have a script in place that automatically kills it.

The preferred way to do background tasks is using cron job. You can set it up using Webmin or `crontab` from SSH.

**Reply by skirodev** [reply]
Sorry I didn't read the documentation carefully, I will change my deployment code as recommended.

Thank you very much for your answer.
